File: 111. Minimum Depth of Binary Tree.py
class Solution1:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        def solve(root: TreeNode | None) -> int:
            if not root:
                return 0
            if not root.left:
                return solve(root.right) + 1
            if not root.right:
                return solve(root.left) + 1
            return min(solve(root.left), solve(root.right)) + 1
        return solve(root)
    
class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        if root is None:
            return 0

        queue = deque([root])
        depth = 0
        while queue:
            depth += 1
            for _ in range(len(queue)):
                node = queue.popleft()
                if node.left is None and node.right is None:
                    return depth
                if node.left: queue.append(node.left)
                if node.right: queue.append(node.right)
            

File: 111. Minimum Depth of Binary Tree.go
func minDepth(root *TreeNode) int {
	if root == nil {return 0}
	if root.Left == nil{return minDepth(root.Right) + 1}
	if root.Right == nil{return minDepth(root.Left) + 1}
	return min(minDepth(root.Left), minDepth(root.Right)) + 1
}